Add mandatory DLEQs to nut-22 (#222)
cashubtc/nuts: 22.md

```json
{
  "id": <hex_str>,
  "secret": <str>,
  "C": <hex_str>,
  "dleq": {
    "e": <str>,
    "s": <str>,
    "r": <str>

>   }

Reviewer: Oops, sorry I think I had a brain hiccup. The wallet should *never* send `r` to the mint because it links the token to its creation. The `r` value is only included in a proof when it's sent from user to user. 

I think we should probably ditch this whole `dleq` field after all and just make sure we validate it when we mint the proof.

Reviewer: Reworded, thank you

`AuthProofs` are single-use. The wallet MUST delete the `AuthProof` after a successful request, and SHOULD delete it even if request results in an error. If the wallet runs out of `AuthProofs`, it can [mint new ones](#minting-blind-authentication-tokens) using its clear authentication token (CAT).

### DLEQs

> To prevent a mint pinning blind authentication token wallets SHOULD check that the returned `AuthProofs` contain a valid DLEQ as defined in [NUT-12](./12.md).

Author: Not sure about the semantics TBH. My thought process was that while it would be better for wallets to check this, it would not break the functionality of the NUT or compatibility with mints / wallets if they don't.

> To prevent a mint pinning blind authentication token wallets SHOULD check that the returned `AuthProofs` contain a valid DLEQ as defined in [NUT-12](./12.md).

Reviewer: ```suggestion
To prevent pinning, wallets MUST validate the DLEQ proofs contained in each `AuthProof` as defined in [NUT-12](./12.md).
```

## Mint

### DLEQs

> The mint MUST return a DLEQ proof with all signatures it returns as defined in [NUT-12](./12.md)

Reviewer: ```suggestion
The mint MUST return DLEQ proofs for every signature it returns as defined in [NUT-12](./12.md)
```
